### ISSUE TITLE
Use `fsspec` to provide a unified interface to GCP, AWS, and local filesystems

### ISSUE THREAD

**Issue by JackKelly** [openclimatefix/nowcasting_dataset#153]
At the moment, our code implements multiple functions to interact with data on GCP vs the local filesystem.  This is exactly the problem that [`fsspec`](https://filesystem-spec.readthedocs.io/en/latest/usage.html) solves:  `fsspec` provides a unified interface to cloud and local storage.  If we use `fsspec` then we can probably throw away quite a lot of our filesystem functions :)

Before starting work on this issue, we should wait for #152 to be merged into `main`.

**Comment by JackKelly** [comment]
I'm making a start on this in #152 (only where necessary to make it easy to write to local filesystem).  I'll leave it to a later PR to complete this task.

**Comment by JackKelly** [comment]
With `num_workers=4`, the script took 11hrs and 38 minutes to output 3,400 batches, which is 12.3 seconds per batch, and `iotop` reports that the total read speed rarely exceeds 50 MB/sec.

With`num_workers=1`, the script took 230 seconds to do 12 batches (19 secs per batch); and total read speeds hover around 30 - 70 MB/sec.  But I should leave this running today and see how long it takes to do more batches, as it is a bit bursty.

**Comment by JackKelly** [comment]
A word of warning:  `Pathy.exists()` seems buggy to me:  `Path('gs://bucket/foo/bar.txt').exists()` sometimes hangs.  `Pathy('/local/file.txt').exists()` sometimes returns the wrong answer (although maybe we're supposed to include a protocol prefix like `'file://local/file.txt'`?)  We should explore this more and, if this really does turn out to be a bug, we should report it to the `Pathy` project.

That said, using `Pathy` as a prettier way to join paths works really well.  e.g. `Pathy('gs://bucket') / 'foo' / 'bar.txt'`
